common/views/components/BaseTabs/BaseTabs.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState, useRef, useContext, KeyboardEvent } from 'react';
+import { AppContext } from '@weco/common/views/components/AppContext/AppContext';
+import styled from 'styled-components';
+import { classNames } from '@weco/common/utils/classnames';
+
+const TabList = styled.div.attrs({
+  role: 'tablist',
+})``;
+
+type TabProps = {
+  isActive: boolean;
+  tabPanelId: string;
+};
+
+const Tab = styled.button.attrs((props: TabProps) => ({
+  className: classNames({
+    'plain-button no-padding': true,
+  }),
+  role: 'tab',
+  tabIndex: props.isActive ? 0 : -1,
+  'aria-selected': props.isActive,
+  'aria-controls': props.tabPanelId,
+}))<TabProps>`
+  &:focus {
+    outline: 0;
+  }
+`;
+
+type TabPanelProps = {
+  id: string;
+  isActive: boolean;
+  isEnhanced: boolean;
+};
+const TabPanel = styled.div.attrs((props: TabPanelProps) => ({
+  id: props.id,
+  role: 'tabpanel',
+  hidden: props.isEnhanced && !props.isActive,
+  'aria-expanded': props.isActive,
+}))<TabPanelProps>``;
+
+export type TabType = {
+  id: string;
+  tab: (isActive: boolean, isFocused: boolean) => JSX.Element;
+  tabPanel: JSX.Element | string;
+};
```

Comment:
This is quite confusing having a `tab` contain a `tab`.

Trying to think of what a tab and it's panel are collectively called?

Reply:
Yeah. I agree. Spoke with @ishouty about this too. But didn't come up with a satisfactory idea.

`tabs` that have a `tabElement` and a `tabPanelElement`? Not really sure that clears much up.

Reply:
[?](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgMQK4DsDGNgXQYQnDwFN14BfOAMyiLgHIoSBDbBgbgFgAoXmAJ5gScACosARgAU6YAM5wAvIjjA5AQWzAAbiQBccCRAgAbVujgV+QkeIlKUGLXkLF0ZGAB47MiPIB81sJiklIs7iYOaFg4LkSQ7uSeCBSBPILBdoTkHg4IvHCFqgAmBnIwUMDoAOYANAVFMJIGdvU8RXBN0uEkJi2hPSa8VnzpNnAAQixyJHZyvvJ5DYVd2TAe-RJrHgDaALrDvJh45ZPTs5IKygAUSKt46+SWBlMzcwtyAJRK-ogjQA)

Reply:
I don't follow

Reply:
Should we be passing react elements, no functions that return react elements? If so you can use the React type system to reflect that.

Reply:
I don't think we can do that and continue to have access to the `isActive` and `isFocused` values in the consuming component – I think the [render prop needs to be a function](https://reactjs.org/docs/render-props.html)